data/lua/wml/random_placement.lua

```diff
@@ -14,7 +14,7 @@ wesnoth.wml_actions.random_placement = function(cfg)
 	local math_abs = math.abs	
 	local locs = wesnoth.get_locations(filter)
 	if type(num_items) == "string" then		
-		num_items = math.floor(loadstring("local size = " .. #locs .. "; return " .. num_items)())
+		num_items = math.floor(load("local size = " .. #locs .. "; return " .. num_items)())
```

Comment:
It still looks like amazingly terrible code. As this is a wml action, passing in a closure or something isn't going to work, but is calling the equivalent of exec() really the best we can come up with?